src/hw9/ArrayValueCalculator.java

```diff
@@ -3,25 +3,25 @@
 public class ArrayValueCalculator{
 
 
-        public static int doCalc(String[][] arr) throws ArraySizeException, ArrayDataException {
-            // Check the size of the array
-            if (arr.length != 4 || arr[0].length != 4) {
-                throw new ArraySizeException("The input array must be of size 4x4");
-            }
+    public static int doCalc ( String[][] arr ) throws ArraySizeException, ArrayDataException{
```

Comment:
`throws ArraySizeException, ArrayDataException` should be used if you wanna delegate error handling to the end user. The description asks you to validate the input on your own, so throws is redundant here, remove it.